tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/CompilableTask.java

```diff
@@ -144,6 +146,26 @@ public void setBatchThreads(long batchThreads) {
         this.batchNumThreads = batchThreads;
     }
 
+    @Override
+    public void setBatchNumber(int batchNumber) {
+        this.batchNumber = batchNumber;
+    }
+
+    @Override
+    public int getBatchNumber() {
+        return this.batchNumber;
+    }
+
+    @Override
+    public void setBatchSize(long batchSize) {
```

Comment:
What is the difference between batchSIze and batchThread

```java
void setBatchThreads(long batchThreads);

void setBatchSize(long batchSize);
```

Reply:
The value of `batchThreads` changes when the chunks are not even, the `batchSize` is always the initial batch size. For example, if we have 3 batches, 60000, 60000 and 20000, the batchSize will be 60000 throughout the computation, even though the batch threads will change from 60000 to 20000 when computing the final chunk.